prow/test/integration/integration-test.sh

```diff
@@ -34,7 +34,7 @@ function setup() {
   "${bazel}" run //prow/test/integration:setup-local-registry "$@" || ( echo "FAILED: set up local registry">&2; return 1 )
 
   if [[ "${PUSH_IMAEG_WITH_KO:-}" == "true" ]]; then
-    PUSH=true KO_DOCKER_REPO="localhost:5001" ./prow/prow-images.sh
+    PUSH=true KO_DOCKER_REPO="localhost:5001" PROW_IMAGES_DEF_FILE="prow/test/integration/prow/.prow-images" ./prow/prow-images.sh
```

Comment:
Do we test that all images can be built elsewhere? I'm pretty sure with bazel [all the images were built as part of presubmit testing](https://github.com/kubernetes/test-infra/blob/8b41e455b4f1cb387da060f597a77bddad0e1014/config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml#L19) so it wasn't possible to submit a change that would break image building. If this is the only place we build the images in presubmit we should probably build them all rather than a subset.

Reply:
good point. The image building process is slow right now, let me make them parallel to speed it up then we can push all

Reply:
actually `bazel test` does image building but not pushing, and pushing is where most of the time is spent as far as I can observe. I'll push the changes that improves ko performance but would like to keep integration test pushing only relevant images, I think we'll probably need a separate job to check image building if that's what we would like to do

Reply:
> I think we'll probably need a separate job to check image building if that's what we would like to do

Until we have some job doing that it isn't safe to switch to using Makefile built containers in production. A separate job or including this in the unit testing job seems fine to me. I'm assuming that `ko` would benefit from Go's local build cache being warmed by building unit tests, making the binary compilation step faster at least.

Reply:
added image building as part of unit test